tests/unit/rate_limiting/test_integration.py

```
@@ -0,0 +1,297 @@
+            await rate_limiter.execute_with_protection(
+                "invalid-model",
+                lambda: "test"
+            )
+    
+    @pytest.mark.asyncio
+    async def test_redis_failure_handling(self, rate_limiter, mock_redis):
+        """Test handling of Redis failures."""
+        # Mock Redis failure
+        mock_redis.pipeline.side_effect = Exception("Redis down")
+        
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)